### ISSUE TITLE
Error trying to open a tab-separated file

### ISSUE THREAD

**Issue by prvst** [alexhallam/tv#96]
Hi, I'm trying to visualize a TSV file, and I'm getting this error:

```
thread 'main' panicked at 'a csv record: Error(UnequalLengths { pos: Some(Position { byte: 513, line: 2, record: 1 }), expected_len: 37, len: 8 })', src/main.rs:312:20
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

I'm using the latest Linux release.

**Comment by alexhallam** [comment]
Thanks! Could you paste the command you used and a small (1-2 line) csv so I can reproduce the problem?

**Comment by Lireer** [comment]
Seems like a good idea to me as long as the effects of different file extensions are documented and setting the delimiter via cli options is easier than changing a files extension. So in my opinion we should check for the extension and parse values like `\t`. Are there other values that should be accepted (see [ANSI-C Quoting](https://www.gnu.org/software/bash/manual/html_node/ANSI_002dC-Quoting.html)) and where do we draw the line of what is expected to work and what isn't. Maybe just start with the extension check and `\t`, others can be added later and until then the user has to figure out how their shell works.

Some thoughts on the design of the API:

Implementing any of this shouldn't be a problem, I'm just wondering what kind of behavior and features do users expect and use. Would they expect a `.tsv` file to automatically use tabs as delimiters? I guess the answer to that is likely yes, but I don't know. They shouldn't feel the need to find some hacky solution, e.g. the shell makes entering the tab byte hard, so they change the extension of their file to be recognized as TSV. 

In general these options configure the reader using different ways. This means there should be an intuitive order of priority. I feel like it's quite easy to make automatic features unintuitive, so we should make sure they are clearly documented. For me the expected behavior would be:

1. Standard csv configuration as the default
1. Maybe a config file
1. Check the file extension for TSV or PSV (pipe-separated values)
    - Are there more formats and what do they change? Just the delimiter?
1. CLI options should take precedence over everything else

**Comment by alexhallam** [comment]
I agree with making the CLI options take precedence. In my experience the most common exception to the single byte delimiters like `,`, `:`, `;`, `|`, etc. is `\t` and that is probably the only exception that makes sense. I am fine ignoring the many other ANSI-C quoting special characters. 

So lets just allow users to pass `-s "\t"` for tab separated files.
